File: missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        for(int i = 0; i < nums.length; i++) {
            int ri = Math.abs(nums[i]);
            if (ri >= nums.length) continue;
            if (nums[ri] == 0) {
                nums[0] = -Math.abs(nums[0]);
                nums[ri] = -nums.length;
            }
            else nums[ri] = -Math.abs(nums[ri]);
        }
        
        for(int i = 0; i < nums.length; i++) {
            if (nums[i] < 0) continue;
            return i;
        }
        
        return nums.length;
    }
}

File: missing-number/missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        
        for i in range(n):
            k = abs(nums[i])
            if k == n + 1:
                k = 0
                
            if k >= n:
                continue
                
            if nums[k] == 0:
                nums[k] = - n - 1
            else:
                nums[k] = -abs(nums[k])
                
        for i, j in enumerate(nums):
            if j >= 0:
                return i
            
        return n